v9.6 Changelog (#6930)
mattermost/docs: source/deploy/mattermost-changelog.md

:depth: 2
```

## Release v9.6 - [Feature Release](https://docs.mattermost.com/upgrade/release-definitions.html#feature-release)

**Release day: March 15, 2024**

### Compatibility
 - Updated minimum required Edge and Chrome versions to 120+.

```{Important}
If you upgrade from a release earlier than v9.5, please read the other [Important Upgrade Notes](https://docs.mattermost.com/upgrade/important-upgrade-notes.html).
```

### Improvements

#### User Interface (UI)
 - Pre-packaged Calls version [v0.24.0](https://github.com/mattermost/mattermost-plugin-calls/releases/tag/v0.24.0).
 - Pre-packaged GitLab plugin version [v1.8.0](https://github.com/mattermost/mattermost-plugin-gitlab/releases/tag/v1.8.0).
 - Added the [Outgoing OAuth Connections](https://mattermost.com/pl/outgoing-oauth-connections) integration type.

>  - Re-designed the **System Console > User Management** screen, and added the ability to batch export users in CSV format.

Reviewer: Should we note that part of this only works on Postgres? 

CC @devinbinnie @harshilsharma63 in case I'm misremembering

Reviewer 2: users can still batch export report on MySQL, just not see the live results in UI IIRC

>  - Re-designed the **System Console > User Management** screen, and added the ability to batch export users in CSV format. On MySQL, users cannot view live results of the batch export in the user interface.

Reviewer 3: Ability to export requires at least a Pro license

>  - Changed notifications so that, if you have **Thread Reply Notifications** disabled for your account and **Automatically follow threads in this channel** enabled for a channel, you'll no longer receive thread notifications for that channel.

Reviewer 4: This is a bit misleading with the `no longer receive thread notifications` given that it wasn't intended to receive thread notifications because the global setting was off. Here is my suggestion, @cwarnermm feel free to make it better: 

```suggestion
- Fixed thread notifications so that if you have **Thread Reply Notifications** disabled for your account and **Automatically follow threads in this channel** enabled for a channel, you won't receive thread notifications for that channel per global setting.
```

Also, @amyblais maybe we should move this to bug fixes instead because it was a bug more than an improvement

Reviewer 5: Thanks, @marianunez - much clearer!

 - Added support for checkbox types in the **System Admin** settings.
 - Added support for WebP image previews in the web app similar to PNG and other image formats.
 - Several pre-packaged plugins were removed.
 - Added the Outgoing OAuth Connections integration type.

#### Administration

>  - Removed some unused Redux actions and reducers.

Reviewer: See https://github.com/mattermost/mattermost/pull/26006. For reducers, there's nothing particularly noteworthy other than maybe `state.entities.posts.selectedPostId` which I was kind of surprised to find we didn't use. I'd be surprised if any of that would affect plugins

:depth: 2
```

## Release v9.6 - [Feature Release](https://docs.mattermost.com/upgrade/release-definitions.html#feature-release)

**Release day: March 15, 2024**

### Compatibility
 - Updated minimum required Edge and Chrome versions to 120+.

```{Important}
If you upgrade from a release earlier than v9.5, please read the other [Important Upgrade Notes](https://docs.mattermost.com/upgrade/important-upgrade-notes.html).
```

### Improvements

#### User Interface (UI)
 - Pre-packaged Calls version [v0.24.0](https://github.com/mattermost/mattermost-plugin-calls/releases/tag/v0.24.0).
 - Pre-packaged GitLab plugin version [v1.8.0](https://github.com/mattermost/mattermost-plugin-gitlab/releases/tag/v1.8.0).
 - Re-designed the **System Console -> User Management** screen, and added a batch CSV export of users.

>  - Changed notifications so that, if you have **Thread Reply Notifications** disabled for your account and "Automatically follow threads in this channel" for a channel, you'll no longer receive thread notifications for that channel.

Reviewer 5: ```suggestion
 - Changed notifications so that, if you have **Thread Reply Notifications** disabled for your account and **Automatically follow threads in this channel** for a channel, you'll no longer receive thread notifications for that channel.
```

Author: @harshilsharma63 @devinbinnie Would you know the answer to this? (For https://github.com/mattermost/mattermost/pull/26062)

Reviewer 6: @cwarnermm @amyblais It should be enabled.

 - Improved the appearance of profile/account menus.

>  - Added support for checkbox types in the **System Admin** settings.

Author: @crspeller @calebroseland Would you know the answer to this question? (for https://github.com/mattermost/mattermost/pull/24151)

>  - Added support for checkbox types in the **System Admin** settings.

Reviewer: I think that's referring to settings in the System Console

 - Added support for webp preview in the web app similar to png and other image formats.

>  - Several pre-packaged plugins were removed.

Author: @sbishel @esethna Do we want to list these?

>  - Added Outgoing OAuth Connections integration type.

Author: @mickmister @fmartingr Would you be able to help with this information?

 - Fixed an issue with extra space getting added to code blocks in search results.
 - Fixed an issue where deactivated members were not included in a favorited **Direct Message** channel export.
 - Fixed an issue where password strength settings wouldn't be disabled if they were set through environment variables.
 - Fixed an issue where post mentions would grow outside the viewport on small devices.
 - Fixed an issue with draft removal after deleting the post.
 - Fixed a markdown issue where, on some occasions, extra space was found before a list.
 - Fixed an issue where a sender to a custom group would also receive the message notification themselves.
 - Fixed a web app crash when a System Admin clicked on a link to a private channel that they were not a member of.
 - Fixed ``ChannelHasBeenCreated`` plugin hook not being called when a group channel was created.

### config.json
 - Multiple setting options were added to ``config.json``. Below is a list of the additions and their default values on install. The settings can be modified in ``config.json``, or the System Console when available.

#### Changes to all plans:
 - Under ``ServiceSettings`` in ``config.json``:

>     - Added ``EnableOutgoingOAuthConnections`` configuration setting for Outgoing OAuth Connections integration type.

Reviewer 7: This is enterprise only. cc: @mickmister

### Compatibility
 - Updated minimum required Edge and Chrome versions to 120+.

```{Important}
If you upgrade from a release earlier than v9.5, please read the other [Important Upgrade Notes](https://docs.mattermost.com/upgrade/important-upgrade-notes.html).
```

### Improvements

#### User Interface (UI)
 - Pre-packaged Calls version [v0.24.0](https://github.com/mattermost/mattermost-plugin-calls/releases/tag/v0.24.0).
 - Pre-packaged GitLab plugin version [v1.8.0](https://github.com/mattermost/mattermost-plugin-gitlab/releases/tag/v1.8.0).
 - Added the [Outgoing OAuth Connections](https://mattermost.com/pl/outgoing-oauth-connections) integration type.
 - Re-designed the **System Console > User Management** screen, and added the ability to batch export users in CSV format (Professional and Enterprise plans). On MySQL, users cannot view live results of the batch export in the user interface.
 - Improved the appearance of profile/account menus.
 - Added support for checkbox types in the **System Console** settings.
 - Added support for WebP image previews in the web app similar to PNG and other image formats.
 - Several pre-packaged plugins were removed.

#### Administration

>  - Removed some unused Redux actions and reducers, including ``state.entities.posts.selectedPostId``.

Reviewer 8: Wondering if any plugins are using this. I suspect not, but we should probably have an investigation/discussion on this sort of thing before executing on it. @hmhealey @M-ZubairAhmed Do you know what this reducer value is primarily used for, and if plugins may have a use case for using it?

Reviewer: You actually asked about this above, but those actions haven't been used in quite a long time, if ever, in the web app. They carried over from the mobile app since it used mattermost-redux before the web app did

### Go Version
 - v9.6 is built with Go ``v1.20.7``.

### Known Issues
 - Draft is lost from the reply box when other user's root post is deleted [MM-56972](https://mattermost.atlassian.net/browse/MM-56972).
 - Users' initial status is not always loaded correctly [MM-56966](https://mattermost.atlassian.net/browse/MM-56966).
 - Status may sometimes get stuck as **Away** or **Offline** in High Availability mode with IP Hash turned off.
 - Searching stop words in quotation marks with Elasticsearch enabled returns more than just the searched terms.
 - Slack import through the CLI fails if email notifications are enabled.
 - Push notifications don't always clear on iOS when running Mattermost in High Availability mode.
 - The Playbooks left-hand sidebar doesn't update when a user is added to a run or playbook without a refresh.
 - If a user isn't a member of a configured broadcast channel, posting a status update might fail without any error feedback. As a temporary workaround, join the configured broadcast channels, or remove those channels from the run configuration.
 
### Contributors
 - 

> 

Reviewer 8: Not sure if this is a placeholder but the section is currently blank

Author: This is a placeholder for now until we get the list on March 1st.
